parsil/src/assembler.rs

```diff
@@ -422,10 +455,10 @@ impl<'a, C: ContextProvider> Assembler<'a, C> {
                         };
                         let bound = Some(self.expression_to_boundary(left)?);
 
-                        if self.secondary_index_bounds.high.is_some() {
-                            self.secondary_index_bounds.high = None;
+                        if bounds.high.is_some() {
```

Comment:
Why can't we compare `bounds` with `bound` as we do above? In some cases (e.g., when they are both compared to a constant), it should be possible to choose one among these 2 bounds, which might make the query proving much more efficient